### DISCUSSION TITLE
closeOnBeforeunload not working properly after upgrading to socketio v4.7.5

### DISCUSSION THREAD

**Discussion by sarmad36** [socketio/socket.io#5055] [Show and tell] [answered]
With following configuration
this.socket = io({
      closeOnBeforeunload: false,
      reconnection: true,
      reconnectionDelay: 100,
      reconnectionDelayMax: 2000,
      reconnectionAttempts: Infinity,
      transports: ["polling"],
      secure: true,
      rejectUnauthorized: false,
      forceNew: false,
      timeout: 6000,
})

when I try to leave the page on Firefox the socket connection is disconnected  and if I try to log the reason for disconnection it says " **transport error** "

**Comment by sarmad36** [comment]
@darrachequesne can you provide any input here?

**Comment by darrachequesne** [comment, marked as answer]
Hi! When setting `closeOnBeforeunload: false`, the Socket instance will indeed emit a `disconnect` event when the user reloads the page on Firefox.

Reference: https://socket.io/docs/v4/client-options/#closeonbeforeunload

**Comment by sarmad36** [comment]
But if I set it to true is silently closes the transport and I am unable to emit any event.
Basically what I want to do is that on javascript "beforeUnload" event I want to emit a socket event it works fine on chrome and other browsers but on firefox no event is being fired.

**Reply by darrachequesne** [reply]
I was not able to reproduce the issue: https://github.com/socketio/socket.io-fiddle/tree/socket.io/discussions/5055

```
$ node server.js
server listening at http://localhost:3000
connect NyI3hlDYDpKmSbt9AAAB
received 'bye'
disconnect NyI3hlDYDpKmSbt9AAAB due to transport close
connect 38pAi8rwORkfnrbPAAAD
received 'bye'
disconnect 38pAi8rwORkfnrbPAAAD due to transport close
connect jxE3dn1g6Stpb-ZNAAAH
```

The `beforeunload` handler:

```js
window.addEventListener("beforeunload", () => {
  console.log("beforeunload");
  socket.emit("bye");
});
```

Same behavior on Chromium (v123) and Firefox (v127) on Ubuntu 22.04.

Am I missing something?
